pkg/master/master.go

```diff
@@ -210,6 +214,11 @@ func setDefaults(c *Config) {
 		}
 		c.PortalNet = portalNet
 	}
+	if c.PublicServicePorts.Size == 0 {
+		// TODO: Currently no way to specify no PublicServicePorts (do we need to allow this?)
```

Comment:
Torn.  On one hand GCE doesn't need this, so it should be disable-able.  On the other hand, disabling this breaks the strictly-nested property of visibility and means that we need to check with cloud provider to determine if it is required.  E.g. if I set visibility to LoadBalancer, AWS cloud provider says "must have a host port", so we fail if no public ports are available.

After writing it out, I am OK to require it for now with a TODO that maybe we can support not using this on clouds that don't need it.